test/Integration/SqlTriggerBindingIntegrationTests.cs

```diff
@@ -163,5 +207,36 @@ private void DeleteProducts(int first_id, int last_id)
                 id += 1;
             }
         }
+
+        private async Task<string> AssertTriggerListenerError(string functionName)
+        {
+            string errorMessage = null;
+            var tcs = new TaskCompletionSource<bool>();
+
+            void OutputHandler(object sender, DataReceivedEventArgs e)
+            {
+                if (errorMessage == null && e.Data?.Contains("Failed to start SQL trigger listener") == true)
+                {
+                    string exceptionPrefix = "Exception: System.InvalidOperationException: ";
+                    int index = e.Data.IndexOf(exceptionPrefix, StringComparison.Ordinal);
+                    Assert.NotEqual(-1, index);
+                    errorMessage = e.Data[(index + exceptionPrefix.Length)..];
+                    tcs.SetResult(true);
+                }
+            };
+
+            this.StartFunctionHost(functionName, Common.SupportedLanguages.CSharp, false, OutputHandler);
+            Task completedTask = await Task.WhenAny(tcs.Task, Task.Delay(TimeSpan.FromSeconds(5)));
```

Comment:
Is there a reason you're not just using `Task.Wait(STARTUP_ERROR_TIMEOUT)` and using the value of that to determine if it succeeded or not instead of doing this + the not null check at the end?

Reply:
@Charles-Gagnon - I am not able to follow this comment. Can you elaborate a little? `Task.Wait` blocks the thread, so instead using `Task.WhenAny()`. Just good practice.

Reply:
Then just use Task.WaitAsync. The point is that this code currently seems a bit convoluted when essentially all you care about is "did our task complete in the expected time period". WaitAsync with a timeout is just a simpler and more straightforward way of doing that in my opinion.

Reply:
`Task.WaitAsync` was my first thought but it's not available in .NET Standard 2.0 ([doc link](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-6.0&viewFallbackFrom=netstandard-2.0)).

Reply:
Using non-async `Task.Wait` as suggested in first comment.

Reply:
That's unfortunate. Task.Wait should be fine for tests - if we see any issues then going back to what you had originally is fine too in this case.